androidapp/HopSquadApp/app/src/main/java/com/hopsquad/hopsquadapp/fragments/TapListFragment.java

```diff
@@ -27,28 +47,94 @@
     private BeerAdapter mAdapter;
     private FloatingActionButton placeOrderBtn;
+    private PaymentsClient mPaymentsClient;
 
     private TapListViewModel viewModel;
 
     public TapListFragment() {
         // Required empty public constructor
     }
 
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        switch (requestCode) {
+            case LOAD_PAYMENT_DATA_REQUEST_CODE:
+                switch (resultCode) {
+                    case Activity.RESULT_OK:
+                        PaymentData paymentData = PaymentData.getFromIntent(data);
+                        // You can get some data on the user's card, such as the brand and last 4 digits
+                        CardInfo info = paymentData.getCardInfo();
+                        // You can also pull the user address from the PaymentData object.
+                        UserAddress address = paymentData.getShippingAddress();
+                        // This is the raw JSON string version of your Stripe token.
+                        String rawToken = paymentData.getPaymentMethodToken().getToken();
+
+                        registerPlacedOrder(rawToken);
+
+                        break;
+                    case Activity.RESULT_CANCELED:
+                        break;
+                    case AutoResolveHelper.RESULT_ERROR:
+                        Status status = AutoResolveHelper.getStatusFromIntent(data);
+                        // Log the status for debugging
+                        // Generally there is no need to show an error to
+                        // the user as the Google Payment API will do that
+                        break;
+                    default:
+                        // Do nothing.
+                }
+                break; // Breaks the case LOAD_PAYMENT_DATA_REQUEST_CODE
+            // Handle any other startActivityForResult calls you may have made.
+            default:
+                // Do nothing.
+        }
+    }
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         viewModel = ViewModelProviders.of(this).get(TapListViewModel.class);
         viewModel.setRepository(new WebServiceRepository());
         viewModel.init();
+
+        initializePaymentsClient();
+    }
+
+    private void initializePaymentsClient() {
+        mPaymentsClient = Wallet.getPaymentsClient(this.getContext(), new Wallet.WalletOptions.Builder()
+                .setEnvironment(WalletConstants.ENVIRONMENT_TEST).build());
+
+        IsReadyToPayRequest request = IsReadyToPayRequest.newBuilder()
+                .addAllowedPaymentMethod(WalletConstants.PAYMENT_METHOD_CARD)
+                .addAllowedPaymentMethod(WalletConstants.PAYMENT_METHOD_TOKENIZED_CARD)
+                .build();
+
+        Task<Boolean> task = mPaymentsClient.isReadyToPay(request);
```

Comment:
Is this how to send the payment request to the server?  I couldn't figure out how in stripe.  They have what to do once the token and payment info is received on the server end, but not sure how to send it correctly.  I know it says you can't process the payment on the mobile app and it has to be done on the server.

Reply:
I think that is implemented differently for the Santandard Stripe integration.